adapters/richaudience/richaudience.go

```diff
@@ -0,0 +1,349 @@
+	richaudienceRequest := richaudienceRequest{}
+
+	raiHeaders := http.Header{}
+	//Global Vars
+	secure := int8(0)
+	//Object IMP
+	resImps := make([]openrtb2.Imp, 0, len(request.Imp))
+	raiErrors := make([]error, 0, len(request.Imp))
+
+	var regsExt *openrtb_ext.ExtRegs
+	if request.Regs != nil && request.Regs.Ext != nil {
+		if err := json.Unmarshal(request.Regs.Ext, &regsExt); err != nil {
+			fmt.Println(err)
+		}
+	}
+
+	if regsExt != nil {
+		fmt.Println(regsExt)
+	}
+
+	setHeaders(&raiHeaders)
+
+	richaudienceRequest.ID = request.ID
+
+	raiImp, err := setImp(request, &secure, &richaudienceRequest)
+
+	if err != nil {
+		fmt.Printf("%s", err)
+	}
+
+	resImps = append(resImps, raiImp)
+	richaudienceRequest.Imp = resImps
+
+	setSite(request, &richaudienceRequest)
+
+	err = setDevice(request, &richaudienceRequest)
+	if err != nil {
+		raiErrors = append(raiErrors, &errortypes.BadInput{
+			Message: err.Error(),
+		})
+		return nil, raiErrors
+	}
+
+	err = setUser(request, &richaudienceRequest)
+	if err != nil {
+		raiErrors = append(raiErrors, &errortypes.BadInput{
+			Message: err.Error(),
+		})
+		return nil, raiErrors
+	}
+
+	//User: consent
+	req, err := json.Marshal(richaudienceRequest)
+	if err != nil {
+		fmt.Printf("%s", err)
+	}
+
+	urlParams := macros.EndpointTemplateParams{Host: "http://ortb.richaudience.com/ortb/?bidder=pbs"}
+	url, err := macros.ResolveMacros(&a.EndpointTemplate, urlParams)
+
+	requestData := &adapters.RequestData{
+		Method:  "POST",
+		Uri:     url,
+		Body:    req,
+		Headers: raiHeaders,
+	}
+
+	return []*adapters.RequestData{requestData}, raiErrors
+}
+
+func (a *RichaudienceAdapter) MakeBids(request *openrtb2.BidRequest, requestData *adapters.RequestData, responseData *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+
+	if responseData.StatusCode == http.StatusNoContent {
+		return nil, nil
+	}
+
+	if responseData.StatusCode == http.StatusBadRequest {
+		err := &errortypes.BadInput{
+			Message: "Unexpected status code: 400. Bad request from publisher. Run with request.debug = 1 for more info.",
+		}
+		return nil, []error{err}
+	}
+
+	var bidResp openrtb2.BidResponse
+	if err := json.Unmarshal(responseData.Body, &bidResp); err != nil {
+		return nil, []error{&errortypes.BadServerResponse{
+			Message: err.Error(),
+		}}
+	}
+
+	if responseData.StatusCode != http.StatusOK {
+		err := &errortypes.BadServerResponse{
+			Message: fmt.Sprintf("Unexpected status code: %d. Run with request.debug = 1 for more info.", responseData.StatusCode),
+		}
+		return nil, []error{err}
+	}
+
+	var response openrtb2.BidResponse
+	if err := json.Unmarshal(responseData.Body, &response); err != nil {
+
+		return nil, []error{err}
+	}
+
+	bidResponse := adapters.NewBidderResponseWithBidsCapacity(len(request.Imp))
+	bidResponse.Currency = response.Cur
+	for _, seatBid := range response.SeatBid {
+		for _, bid := range seatBid.Bid {
+			b := &adapters.TypedBid{
+				Bid:     &bid,
+				BidType: "banner",
```

Comment:
Based on `static/bidder-info/richaudience.yaml` file you support banner and video imps. Here you expect only banner bids. This should be fixed here or in `richaudience.yaml` file to avoid ambiguity.

Reply:
in next commit i have solved:

for the moment only we will only have demand the banner.